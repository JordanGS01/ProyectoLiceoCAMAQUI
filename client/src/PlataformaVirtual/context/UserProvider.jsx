

import { useState } from 'react'
import { UserContext } from './UserContext'


const userDataInitialState = {
    nombre: '',
    cedula: 0,
    tipo: ''
}

export const UserProvider = ({ children }) => {

    const [userData, setUserData] = useState(userDataInitialState);
    const [loged, setLoged] = useState(false);

    const setUser = ({ nombre, cedula, tipo }) => {
        setUserData({
            nombre,
            cedula,
            tipo
        })
        setLoged( true );
    }

    const logOutUser = () => {
        setUserData(userDataInitialState);
        setLoged( false );
    }

    return (
        <UserContext.Provider value={{ 
            userData,
            setUser,
            logOutUser,
            loged
        }}>
            { children }
        </UserContext.Provider>
    )
}