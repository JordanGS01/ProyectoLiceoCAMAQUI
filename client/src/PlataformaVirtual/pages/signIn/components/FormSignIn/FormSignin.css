

.FormSignin-ContenedorBotones{
    display: flex;
    flex-direction: row-reverse;
}