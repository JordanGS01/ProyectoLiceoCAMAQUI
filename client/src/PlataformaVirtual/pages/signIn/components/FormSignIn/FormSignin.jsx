

import { useState } from 'react';
import { useNavigate } from 'react-router';

import { Box, FormControl, TextField, Button, InputAdornment, IconButton, Typography } from '@mui/material'
import { PermIdentityOutlined ,BadgeOutlined, PasswordOutlined, Visibility, VisibilityOff } from '@mui/icons-material'

import './FormSignin.css'
import { useForm } from '../../../../hooks'
import { onFormSubmit } from '../../helpers/registrarUsuario'

import { Alert } from '../../../../../ui/components/Alert';


export const FormSignin = () => {
  
    const [showPassword, setShowPassword] = useState(false);
    const [warnings, setWarnings] = useState({
        nombreInvalido: false,
        cedulaInvalida: false,
        contraInvalida: false,
        contraNoCoincide: false
    })
    const [openAlertSuccess, setOpenAlertSuccess] = useState(false);
    const [openAlertError, setOpenAlertError] = useState(false);

    const navigate = useNavigate();

    const { formState, onInputChange } = useForm({
        nombre: '',
        cedula: '',
        contra: '',
        repContra: '',
        tipo: 'E'
    });
    const { nombre, cedula, contra, repContra } = formState;
    const { nombreInvalido, cedulaInvalida, contraInvalida, contraNoCoincide } = warnings;


    return (
    <Box
        component="span" 
        maxWidth='sm'
        sx={{   p: 2,
                border: '1px solid grey',
                background: '#4FA4D3',
                borderRadius: '20px',
                display: 'flex',
                flexDirection: 'column'
            }}
    >
        <form onSubmit={ (e) => { onFormSubmit( e, formState, setWarnings, setOpenAlertSuccess, setOpenAlertError ) } }>

            <Typography variant="h3" gutterBottom sx={{ ml: 2, color: '#FFFFFF' }}>
                Registrarse
            </Typography>
            
            {/* Sección de nombre completo */}
            <FormControl sx={{ m: 2, ml: 3, width: '90%' }}>
                <TextField
                    required
                    autoFocus
                    label='Nombre completo'
                    id='nombre-usuario'
                    InputProps={{
                        startAdornment:(
                            <InputAdornment position='start'>
                                <PermIdentityOutlined />
                            </InputAdornment>
                        )
                    }}
                    sx={{ backgroundColor: '#FFFDFD', borderRadius: '5px' }}
                    variant='filled'
                    

                    { ... ( nombreInvalido && { error: true } ) }
                    { ... ( nombreInvalido && { helperText: 'El nombre no debe contener números o símbolos' } ) }

                    value={nombre}
                    name='nombre'
                    onChange={onInputChange}
                />
            </FormControl>

            {/* Sección de la cédula */}
            <FormControl sx={{ m: 2, ml: 3, width: '90%' }}>
                <TextField
                    required
                    label='Número de cédula'
                    id='cedula-usuario'
                    InputProps={{
                        startAdornment:(
                            <InputAdornment position='start'>
                                <BadgeOutlined />
                            </InputAdornment>
                        )
                    }}
                    sx={{ backgroundColor: '#FFFDFD', borderRadius: '5px' }}
                    variant='filled'
                    type='number'

                    { ... ( cedulaInvalida && { error: true } ) }
                    { ... ( cedulaInvalida && { helperText: 'La cédula debe conener 9 dígitos' } ) }

                    value={cedula}
                    name='cedula'
                    onChange={onInputChange}
                />
            </FormControl>
            
            {/* Sección de la contraseña */}
            <FormControl sx={{ m: 2, ml: 3, width: '90%' }}>
                <TextField
                    required
                    label='Contraseña'
                    id='contra-usuario'
                    type={ showPassword ? 'text' : 'password' }
                    InputProps={{
                        startAdornment:(
                            <InputAdornment position='start'>
                                <PasswordOutlined />
                            </InputAdornment>
                        ),
                        endAdornment:(
                            <InputAdornment position='end'>
                                <IconButton
                                    aria-label='toggle password visibility'
                                    onClick={ () => setShowPassword(!showPassword) }
                                    onMouseDown={ (event) => event.preventDefault() }
                                >
                                    { showPassword ? <VisibilityOff /> : <Visibility /> }
                                </IconButton>
                            </InputAdornment>
                        )
                    }}
                    sx={{ backgroundColor: '#FFFDFD', borderRadius: '5px' }}
                    variant='filled'

                    { ... ( contraInvalida && { error: true } ) }
                    { ... ( contraInvalida && { helperText: 'La contraseña debe ser de mínimo 6 dígitos' } ) }

                    value={contra}
                    name='contra'
                    onChange={onInputChange}
                />

            </FormControl>

            {/* Sección de confirmar contraseña */}
            <FormControl sx={{ m: 2, ml: 3, width: '90%' }}>
                <TextField
                    required
                    label='Vuelva a ingresar la contraseña'
                    id='repetir-contra-usuario'
                    type={ showPassword ? 'text' : 'password' }
                    InputProps={{
                        startAdornment:(
                            <InputAdornment position='start'>
                                <PasswordOutlined />
                            </InputAdornment>
                        ),
                        endAdornment:(
                            <InputAdornment position='end'>
                                <IconButton
                                    aria-label='toggle password visibility'
                                    onClick={ () => setShowPassword(!showPassword) }
                                    onMouseDown={ (event) => event.preventDefault() }
                                >
                                    { showPassword ? <VisibilityOff /> : <Visibility /> }
                                </IconButton>
                            </InputAdornment>
                        )
                    }}
                    sx={{ backgroundColor: '#FFFDFD', borderRadius: '5px' }}
                    variant='filled'

                    { ... ( contraNoCoincide && { error: true } ) }
                    { ... ( contraNoCoincide && { helperText: 'Las contraseñas no coinciden' } ) }

                    value={repContra}
                    name='repContra'
                    onChange={onInputChange}
                />

            </FormControl>
            
            {/* Sección de la botones */}
            <div className='FormSignin-ContenedorBotones'>

                <Button
                    variant="contained"
                    sx={{ ml: 1, mr: 3 }}
                    type='submit'
                >
                    Registrarse
                </Button>

                <Button
                    variant="contained"
                    onClick={ () => { navigate('/login') } }
                >
                    Regresar
                </Button>

            </div>
    
        </form>
        
        {/* Alert que se mostrará cuando el registro se culmine exitosamente */}
        <Alert 
            open = { openAlertSuccess }
            handleClose = {() => { setOpenAlertSuccess(false) }}
            title = "Registro exitoso"
            content = "Se ha registrado exitosamente en el sistema"
            acceptButtonText = "Aceptar"
            acceptButtonFunction = { () => { navigate('/login') } }
            oneButton = { true }
        />
        
        {/* Alert que se mostrará cuando no se pueda registrar el usuario porque ya 
            existe otro usuario con el número de cédula ingresado */}
        <Alert 
            open = { openAlertError }
            handleClose = {() => { setOpenAlertError(false) }}
            title = "Ha ocurrido un error"
            content = "El número de cédula que digitó ya existe registrado en el sistema"
            acceptButtonText = "Aceptar"
            acceptButtonFunction = {() => { setOpenAlertError(false) }}
            oneButton = { true }
        />

    </Box>
  )
}
