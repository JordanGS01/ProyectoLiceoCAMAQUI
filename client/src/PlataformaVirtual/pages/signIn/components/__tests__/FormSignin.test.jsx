

import { cleanup, render, screen, fireEvent } from '@testing-library/react'

import { BrowserRouter } from 'react-router-dom' 

import { UserContext } from '../../../../context/UserContext'

import { FormSignin } from '../FormSignIn/FormSignin'


describe('Pruebas sobre el componente <FormLogin />', () => { 
    
    let containerFormSignIn;

    beforeEach(() => {
        const userData = {};
        const setUser = () => {};
        const logOutUser = () => {};
        const loged = false;
        const isStudent = () => {};
        const isProfessor = () => {};
        const isAdmin = () => {};

        const { container } = render(
            <BrowserRouter>
                <UserContext.Provider value={{ 
                    userData, setUser, logOutUser,
                    loged, isStudent, isProfessor,
                    isAdmin
                }}>
                    <FormSignin />
                </UserContext.Provider>
            </BrowserRouter>
        );

        containerFormSignIn = container;
    })

    
    test('Debe hacer match con el snapshot', () => {
        expect( containerFormSignIn ).toMatchSnapshot();
    })

    test('El valor del input nombre completo debe cambiar', () => { 
        const nombre = 'Jordán Guzmán Salas';

        const inputNombre = screen.getByRole('textbox', { name: 'Nombre completo' });
        expect(inputNombre.value).toEqual('');
        fireEvent.input(inputNombre, { target: { value: nombre } });
        expect(inputNombre.value).toEqual(`${nombre}`);
    })

    test('El valor del input cedula debe cambiar', () => { 
        const cedula = 208140251;

        const inputCedula = screen.getByRole('spinbutton', { name: 'Número de cédula' });
        expect(inputCedula.value).toEqual('');
        fireEvent.input(inputCedula, { target: { value: cedula } });
        expect(inputCedula.value).toEqual(`${cedula}`);
    })

    test('El valor del input contraseña debe cambiar', () => { 
        const contra = "contraSegura";

        const botonVisibilidad = screen.getAllByRole('button', { name: 'toggle password visibility' })[0];
        fireEvent.click(botonVisibilidad);
        
        const inputContra = screen.getByRole('textbox', { name: 'Contraseña' });
        expect(inputContra.value).toEqual('');
        fireEvent.input(inputContra, { target: { value: contra } });
        expect(inputContra.value).toEqual(`${contra}`);
    })

    test('El valor del input repContra debe cambiar', () => { 
        const contra = "contraSegura";

        const botonVisibilidad = screen.getAllByRole('button', { name: 'toggle password visibility' })[0];
        fireEvent.click(botonVisibilidad);
        
        const inputContra = screen.getByRole('textbox', { name: 'Vuelva a ingresar la contraseña' });
        expect(inputContra.value).toEqual('');
        fireEvent.input(inputContra, { target: { value: contra } });
        expect(inputContra.value).toEqual(`${contra}`);
    })

    test('La contraseña debe ser visible al dar click en el botón de visibilidad', () => { 
        const botonVisibilidad = screen.getAllByRole('button', { name: 'toggle password visibility' })[0];
        
        fireEvent.click(botonVisibilidad);
        expect(screen.getByRole('textbox', { name: 'Contraseña' })).toBeTruthy();
    })

    test('Al ingresar mal el nombre, la cedula, contraseña y la repContra, debe informarse al usuario', () => { 
        const nombre = 'Jordán123213';
        const cedula = 2081402222;
        const contra = 'asdqw'
        const repContra = 'asdasd'

        const inputNombre = screen.getByRole('textbox', { name: 'Nombre completo' });
        fireEvent.input(inputNombre, { target: { value: nombre } });

        const inputCedula = screen.getByRole('spinbutton', { name: 'Número de cédula' });
        fireEvent.input(inputCedula, { target: { value: cedula } });

        const botonVisibilidad = screen.getAllByRole('button', { name: 'toggle password visibility' })[0];
        fireEvent.click(botonVisibilidad);
        
        const inputContra = screen.getByRole('textbox', { name: 'Contraseña' });
        fireEvent.input(inputContra, { target: { value: contra } });

        const inputRepContra = screen.getByRole('textbox', { name: 'Vuelva a ingresar la contraseña' });
        fireEvent.input(inputRepContra, { target: { value: repContra } });

        const botonRegistrarse = screen.getByRole('button', {name: 'Registrarse'});
        fireEvent.click(botonRegistrarse);

        expect(screen.getByText('El nombre no debe contener números o símbolos')).toBeTruthy();
        expect(screen.getByText('La cédula debe conener 9 dígitos')).toBeTruthy();
        expect(screen.getByText('La contraseña debe ser de mínimo 6 dígitos')).toBeTruthy();
        expect(screen.getByText('Las contraseñas no coinciden')).toBeTruthy();
    })


    afterEach(() => {
        cleanup();
    })

})