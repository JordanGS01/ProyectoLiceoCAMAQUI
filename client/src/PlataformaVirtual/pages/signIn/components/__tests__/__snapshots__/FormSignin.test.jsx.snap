// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <FormLogin /> Debe hacer match con el snapshot 1`] = `
<div>
  <span
    class="MuiBox-root css-s5di19"
  >
    <form>
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-rlscyn-MuiTypography-root"
      >
        Registrarse
      </h3>
      <div
        class="MuiFormControl-root css-1um02wf-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mcjkg0-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="nombre-usuario"
            id="nombre-usuario-label"
          >
            Nombre completo
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-adornedStart css-1xab92c-MuiInputBase-root-MuiFilledInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PermIdentityOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 9c2.7 0 5.8 1.29 6 2v1H6v-.99c.2-.72 3.3-2.01 6-2.01m0-11C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
              id="nombre-usuario"
              name="nombre"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1um02wf-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mcjkg0-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="cedula-usuario"
            id="cedula-usuario-label"
          >
            Número de cédula
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1xab92c-MuiInputBase-root-MuiFilledInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="BadgeOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 12h4v1.5h-4zm0 3h4v1.5h-4z"
                />
                <path
                  d="M20 7h-5V4c0-1.1-.9-2-2-2h-2c-1.1 0-2 .9-2 2v3H4c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm-9 0V4h2v5h-2V7zm9 13H4V9h5c0 1.1.9 2 2 2h2c1.1 0 2-.9 2-2h5v11z"
                />
                <circle
                  cx="9"
                  cy="13.5"
                  r="1.5"
                />
                <path
                  d="M11.08 16.18c-.64-.28-1.34-.43-2.08-.43s-1.44.15-2.08.43c-.56.24-.92.78-.92 1.39V18h6v-.43c0-.61-.36-1.15-.92-1.39z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
              id="cedula-usuario"
              name="cedula"
              required=""
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1um02wf-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mcjkg0-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="contra-usuario"
            id="contra-usuario-label"
          >
            Contraseña
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1cxkmns-MuiInputBase-root-MuiFilledInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PasswordOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 17h20v2H2v-2zm1.15-4.05L4 11.47l.85 1.48 1.3-.75-.85-1.48H7v-1.5H5.3l.85-1.47L4.85 7 4 8.47 3.15 7l-1.3.75.85 1.47H1v1.5h1.7l-.85 1.48 1.3.75zm6.7-.75 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H15v-1.5h-1.7l.85-1.47-1.3-.75L12 8.47 11.15 7l-1.3.75.85 1.47H9v1.5h1.7l-.85 1.48zM23 9.22h-1.7l.85-1.47-1.3-.75L20 8.47 19.15 7l-1.3.75.85 1.47H17v1.5h1.7l-.85 1.48 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H23v-1.5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1o2n4nu-MuiInputBase-input-MuiFilledInput-input"
              id="contra-usuario"
              name="contra"
              required=""
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1um02wf-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mcjkg0-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled css-o943dk-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="repetir-contra-usuario"
            id="repetir-contra-usuario-label"
          >
            Vuelva a ingresar la contraseña
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1cxkmns-MuiInputBase-root-MuiFilledInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="PasswordOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M2 17h20v2H2v-2zm1.15-4.05L4 11.47l.85 1.48 1.3-.75-.85-1.48H7v-1.5H5.3l.85-1.47L4.85 7 4 8.47 3.15 7l-1.3.75.85 1.47H1v1.5h1.7l-.85 1.48 1.3.75zm6.7-.75 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H15v-1.5h-1.7l.85-1.47-1.3-.75L12 8.47 11.15 7l-1.3.75.85 1.47H9v1.5h1.7l-.85 1.48zM23 9.22h-1.7l.85-1.47-1.3-.75L20 8.47 19.15 7l-1.3.75.85 1.47H17v1.5h1.7l-.85 1.48 1.3.75.85-1.48.85 1.48 1.3-.75-.85-1.48H23v-1.5z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1o2n4nu-MuiInputBase-input-MuiFilledInput-input"
              id="repetir-contra-usuario"
              name="repContra"
              required=""
              type="password"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="FormSignin-ContenedorBotones"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-ibwcft-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Registrarse
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Regresar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </span>
</div>
`;
